src/Common/src/CoreLib/System/Globalization/DateTimeFormatInfo.cs

```diff
@@ -414,6 +416,10 @@ public Calendar Calendar
             }
             set
             {
+                if (GlobalizationMode.Invariant)
+                {
+                    throw new PlatformNotSupportedException();
```

Comment:
>throw new PlatformNotSupportedException(); [](start = 20, length = 42)

I don't think this is valid. the user can still set the Gregorian calendar and this should be allowed.

Reply:
This again was a size and dependency concern.  If you allow _any_ calendar to be set here, then the `OptionalCalendars` property will be accessed - which in turn calls `CultureData.CalendarIds`.

My primary goal was to break down those cross-dependencies as much as possible in Invariant mode - and eventually also reduce some of that `CultureData` code.

Reply:
can't this condition written as 

```C#
if (GlobalizationMode.Invariant && !(value is GregorianCalendar))
```

does this will still will not allow break the cross dependencies?